source/shared/cpp/ObjectModel/Util.cpp

```diff
@@ -76,43 +76,116 @@ std::string ValidateColor(const std::string& backgroundColor,
 }
 
 void ValidateUserInputForDimensionWithUnit(const std::string &unit, const std::string &requestedDimension,
```

Comment:
any reason we wouldn't want to use a regular expression here? seems like it would be simpler...

Reply:
i had the same thought after I finished it :-)

---
In reply to: [217201334](https://github.com/Microsoft/AdaptiveCards/pull/1872#discussion_r217201334) [](ancestors = 217201334)

Reply:
changed it to use regex

---
In reply to: [217207111](https://github.com/Microsoft/AdaptiveCards/pull/1872#discussion_r217207111) [](ancestors = 217207111,217201334)